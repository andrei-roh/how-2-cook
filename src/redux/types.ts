export const SET_USER = 'SET_USER';
export const SET_LOADING = 'SET_LOADING';
export const SET_ERROR = 'SET_ERROR';
export const ADD_RECIPES_TO_LIST = 'ADD_RECIPES_TO_LIST';
export const UPDATE_RECIPES_LIST = 'UPDATE_RECIPES_LIST';
export const ADD_IMAGES_TO_LIST = 'ADD_IMAGES_TO_LIST';
export const SET_RECIPE_SEARCH_INPUT = 'SET_RECIPE_SEARCH_INPUT';
export const SET_MANAGE_RECIPES_SCROLL_SIZE = 'SET_MANAGE_RECIPES_SCROLL_SIZE';
export const SET_MANAGE_RECIPES_SCROLL_DIRECTION =
  'SET_MANAGE_RECIPES_SCROLL_DIRECTION';
export const SET_SHOW_MENU = 'SET_SHOW_MENU';
export const ADD_INGREDIENTS_TO_LIST = 'ADD_INGREDIENTS_TO_LIST';
export const SET_INGREDIENT_SEARCH_INPUT = 'SET_INGREDIENT_SEARCH_INPUT';
export const UPDATE_INGREDIENTS_LIST = 'UPDATE_INGREDIENTS_LIST';


