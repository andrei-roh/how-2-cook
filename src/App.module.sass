@use 'src/style/colors'

.app 
  padding-top: 46px
  background-color: colors.$white
  height: 100vh
  box-sizing: border-box
  display: flex
  flex-direction: column
  align-items: center

