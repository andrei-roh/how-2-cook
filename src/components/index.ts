export { Header } from './Header/Header';
export { Button } from './Button/Button';
export { Input } from './Input/Input';
export { Modal } from './Modal/Modal';
export { Notification } from './Notification/Notification';
export { ImageLoader } from './ImageLoader/ImageLoader';
export { TextArea } from './TextArea/TextArea';
export { Select } from './Select/Select';