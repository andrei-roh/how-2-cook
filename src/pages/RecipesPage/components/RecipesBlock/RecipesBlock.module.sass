@use 'src/style/colors'
@import 'src/style/typography'

.recipesServiceWrapper
    display: flex
    flex-direction: column
    width: 100%
    margin: calc(100vh / 3 - 100px) 0 24px 0
    gap: 12px

    .recipesBlockLogo
        margin-top: 140px
        height: 200px

    .emptyMessage
        @extend .T1
        text-align: center

.recipesBlockWrapper
    display: flex
    flex-direction: column
    width: 100%
    margin: 150px 0 24px 0
    gap: 12px
    justify-content: center
    align-items: center
