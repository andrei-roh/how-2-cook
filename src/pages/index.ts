export { TitlePage } from './TitlePage/TitlePage';
export { RecipesPage } from './RecipesPage/RecipesPage';
export { CreateRecipePage } from './CreateRecipePage/CreateRecipePage';
export { ShowRecipePage } from './ShowRecipePage/ShowRecipePage';
export { EditRecipePage } from './EditRecipePage/EditRecipePage';
export { IngredientsPage } from './IngredientsPage/IngredientsPage';
export { WeekPlanningPage } from './WeekPlanningPage/WeekPlanningPage';
export { MyPlansPage } from './MyPlansPage/MyPlansPage';