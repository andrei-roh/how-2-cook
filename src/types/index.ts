import { Action } from '@reduxjs/toolkit';

export interface IUser {
  email: string;
}

export enum DishType {
  Main = 'main',
  Soup = 'soup',
  Dessert = 'dessert',
  Salat = 'salat',
  Snack = 'snack',
}

export type RecipeImage = {
  [key: string]: string;
};

export interface IIngredient {
  id: string;
  name: string;
}

export interface IRecipe {
  id: string;
  imageUrl: string;
  name: string;
  type: DishType;
  isVegan: boolean;
  ingredients: string[];
  description: string;
  composition?: string;
  createdAt?: string;
  createdBy?: string;
  updatedAt?: string;
  updatedBy?: string;
}

export interface IPlanningList {
  id: string;
  name: string;
  ingredientsList: string[];
  recipesList: string[];
}

export interface IState {
  user: IUser;
  loading: boolean;
  error: null | string;
  recipesList: IRecipe[];
  imagesList: RecipeImage[];
  recipesPage: {
    recipeSearchInput: string;
    scrollSize: number;
    scrollDirection: ScrollDirection;
  };
  isMenu: boolean;
  ingredientsList: IIngredient[];
  ingredientsPage: {
    ingredientSearchInput: string;
  }
  previousRoute: string;
  planningPage: {
    ingredientsList: string[];
    recipesList: IRecipe[];
  }
}

export interface IAction extends Action {
  type: string;
  payload: never;
}

export interface IValidationError {
  [key: string]: string;
}

export enum FirebaseErrorType {
  InvalidEmail = 'auth/invalid-email',
  InvalidPassword = 'auth/invalid-credential',
  TooManyRequsts = 'auth/too-many-requests',
}

export enum ImageType {
  Webp = 'image/webp',
  Jpeg = 'image/jpeg',
  Jpg = 'image/jpg',
  Png = 'image/png',
}

export interface IOption {
  name: string;
  value: string;
}

export enum ScrollDirection {
  Up = 'up',
  Down = 'down',
}

export enum Severity {
  Success = 'success',
  Info = 'info',
  Warning = 'warning',
  Error = 'error',
}

export interface ICurrentRecipe extends Partial<IRecipe> {}
