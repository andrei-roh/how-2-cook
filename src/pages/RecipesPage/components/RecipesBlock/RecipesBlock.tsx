import css from './RecipesBlock.module.sass';
import { IRecipe } from 'src/types';
import Fridge from 'src/assets/hand-drawn-food.svg';
import Pancakes from 'src/assets/pancakes.svg';
import { RecipeCard } from '../RecipeCard/RecipeCard';
import Typography from '@mui/material/Typography';
import Stack from '@mui/material/Stack';

interface RecipesBlockProps {
  recipes: IRecipe[];
  isSearch: boolean;
}

export const RecipesBlock = ({ recipes, isSearch }: RecipesBlockProps) => {
  if (recipes.length === 0) {
    return (
      <Stack
        direction='column'
        useFlexGap
        spacing={2}
        className={css.recipesServiceWrapper}
      >
        {isSearch ? (
          <>
            <img className={css.recipesBlockLogo} src={Pancakes} />
            <Typography className={css.emptyMessage}>
              Рецепты не найдены
            </Typography>
          </>
        ) : (
          <>
            <img className={css.recipesBlockLogo} src={Fridge} />
            <Typography className={css.emptyMessage}>
              Добавьте первый рецепт
            </Typography>
          </>
        )}
      </Stack>
    );
  }

  return (
    <Stack
      direction='column'
      useFlexGap
      spacing={2}
      className={css.recipesBlockWrapper}
    >
      {recipes.map((recipe) => (
        <RecipeCard key={recipe.id} {...recipe} />
      ))}
    </Stack>
  );
};
