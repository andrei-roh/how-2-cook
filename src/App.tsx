import { Outlet } from 'react-router-dom';
import css from './App.module.sass';
import { Header, Notification } from './components';

const App = () => {
  return (
    <>
      <Header />
      <div className={css.app}>
        <Outlet />
      </div>
      <Notification />
    </>
  );
};

export default App;
