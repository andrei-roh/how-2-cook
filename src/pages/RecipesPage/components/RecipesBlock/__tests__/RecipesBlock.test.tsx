import { describe, expect, it, vi } from 'vitest';
import { render } from '@testing-library/react';
import { RecipesBlock } from '../RecipesBlock';
import { DishType, IRecipe, IUser } from 'src/types';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

const navigateMock = vi.fn();
const dispatchMock = vi.fn();
const userMock = { email: 'Test@email.com' };
const imagesListMock = [
  {
    id: 'test_image_src_1',
    value: 'test/image/src/1',
  },
  {
    id: 'test_image_src_2',
    value: 'test/image/src/2',
  },
  {
    id: 'test_image_src_3',
    value: 'test/image/src/3',
  },
];
const recipesMock: IRecipe[] = [
  {
    id: '00000000-0000-0000-0000-000000000001',
    imageUrl: 'test_image_src_1',
    name: 'Test Recipe 1',
    type: DishType.Fish,
    ingredients: 'Test Ingredients 1',
    description: 'Test Description 1',
  },
  {
    id: '00000000-0000-0000-0000-000000000002',
    imageUrl: 'test_image_src_2',
    name: 'Test Recipe 2',
    type: DishType.Meat,
    ingredients: 'Test Ingredients 2',
    description: 'Test Description 2',
  },
  {
    id: '00000000-0000-0000-0000-000000000003',
    imageUrl: 'test_image_src_3',
    name: 'Test Recipe 3',
    type: DishType.Vegan,
    ingredients: 'Test Ingredients 3',
    description: 'Test Description 3',
  },
];

const stateMock = {
  user: userMock as IUser,
  loading: false,
  error: null,
  imagesList: imagesListMock,
};

const reducerMock = (state = stateMock) => {
  return state;
};

const storeMock = configureStore({ reducer: reducerMock });

vi.mock('react-router-dom', async () => {
  const actual = await vi.importActual('react-router-dom');

  return {
    ...actual,
    useNavigate: () => ({ navigate: navigateMock }),
  };
});

vi.mock('react-redux', async () => {
  const actual = await vi.importActual('react-redux');

  return {
    ...actual,
    useDispatch: () => ({ dispatch: dispatchMock }),
  };
});

vi.mock('src/utils/getRecipeImage');

describe('Testing: RecipesBlock', () => {
  it('should render', () => {
    const { container } = render(
      <Provider store={storeMock}>
        <RecipesBlock recipes={recipesMock} isSearch={false} />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it('should render if there are no recipes', () => {
    const { container } = render(
      <Provider store={storeMock}>
        <RecipesBlock recipes={[]} isSearch={false} />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it('should render if have not found recipe', () => {
    const { container } = render(
      <Provider store={storeMock}>
        <RecipesBlock recipes={[]} isSearch={true} />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
