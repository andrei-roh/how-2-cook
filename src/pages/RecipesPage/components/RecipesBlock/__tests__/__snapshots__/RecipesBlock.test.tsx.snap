// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing: RecipesBlock > should render 1`] = `
<div>
  <div
    class="MuiStack-root _recipesBlockWrapper_4b0ee0 css-1ucnu3w-MuiStack-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root _recipeCardWrapper_a06f98 css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiStack-root _recipeContainer_a06f98 css-nen11g-MuiStack-root"
      >
        <img
          class="_recipeCardImage_a06f98"
          id="00000000-0000-0000-0000-000000000001"
          src="test/image/src/1"
        />
        <div
          class="MuiStack-root _recipeData_a06f98 css-nen11g-MuiStack-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap _recipeName_a06f98 css-fvtpii-MuiTypography-root"
          >
            Test Recipe 1
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 _recipeType_a06f98 css-ahj2mt-MuiTypography-root"
          >
            Основное
          </p>
          <div
            class="MuiStack-root _buttonsPanel_a06f98 css-nen11g-MuiStack-root"
          >
            <img
              alt="Edit Recipe Button"
              class="_recipeCardLogo_a06f98"
              src="/src/assets/pencil.svg"
            />
            <img
              alt="Show Recipe Button"
              class="_recipeCardLogo_a06f98"
              src="/src/assets/eye.svg"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root _recipeCardWrapper_a06f98 css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiStack-root _recipeContainer_a06f98 css-nen11g-MuiStack-root"
      >
        <img
          class="_recipeCardImage_a06f98"
          id="00000000-0000-0000-0000-000000000002"
          src="test/image/src/2"
        />
        <div
          class="MuiStack-root _recipeData_a06f98 css-nen11g-MuiStack-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap _recipeName_a06f98 css-fvtpii-MuiTypography-root"
          >
            Test Recipe 2
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 _recipeType_a06f98 css-ahj2mt-MuiTypography-root"
          >
            Десерт
            <img
              src="/src/assets/vegetarian.svg"
              width="12"
            />
          </p>
          <div
            class="MuiStack-root _buttonsPanel_a06f98 css-nen11g-MuiStack-root"
          >
            <img
              alt="Edit Recipe Button"
              class="_recipeCardLogo_a06f98"
              src="/src/assets/pencil.svg"
            />
            <img
              alt="Show Recipe Button"
              class="_recipeCardLogo_a06f98"
              src="/src/assets/eye.svg"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root _recipeCardWrapper_a06f98 css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiStack-root _recipeContainer_a06f98 css-nen11g-MuiStack-root"
      >
        <img
          class="_recipeCardImage_a06f98"
          id="00000000-0000-0000-0000-000000000003"
          src="test/image/src/3"
        />
        <div
          class="MuiStack-root _recipeData_a06f98 css-nen11g-MuiStack-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap _recipeName_a06f98 css-fvtpii-MuiTypography-root"
          >
            Test Recipe 3
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 _recipeType_a06f98 css-ahj2mt-MuiTypography-root"
          >
            Салат
          </p>
          <div
            class="MuiStack-root _buttonsPanel_a06f98 css-nen11g-MuiStack-root"
          >
            <img
              alt="Edit Recipe Button"
              class="_recipeCardLogo_a06f98"
              src="/src/assets/pencil.svg"
            />
            <img
              alt="Show Recipe Button"
              class="_recipeCardLogo_a06f98"
              src="/src/assets/eye.svg"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing: RecipesBlock > should render if have not found recipe 1`] = `
<div>
  <div
    class="MuiStack-root _recipesServiceWrapper_4b0ee0 css-1ucnu3w-MuiStack-root"
  >
    <img
      class="_recipesBlockLogo_4b0ee0"
      src="/src/assets/pancakes.svg"
    />
    <p
      class="MuiTypography-root MuiTypography-body1 _emptyMessage_4b0ee0 css-ahj2mt-MuiTypography-root"
    >
      Рецепты не найдены
    </p>
  </div>
</div>
`;

exports[`Testing: RecipesBlock > should render if there are no recipes 1`] = `
<div>
  <div
    class="MuiStack-root _recipesServiceWrapper_4b0ee0 css-1ucnu3w-MuiStack-root"
  >
    <img
      class="_recipesBlockLogo_4b0ee0"
      src="/src/assets/hand-drawn-food.svg"
    />
    <p
      class="MuiTypography-root MuiTypography-body1 _emptyMessage_4b0ee0 css-ahj2mt-MuiTypography-root"
    >
      Добавьте первый рецепт
    </p>
  </div>
</div>
`;
